### ISSUE TITLE
What do all those variables in vars.vault.yml mean? (Beginners trouble)

### ISSUE THREAD

**Issue by DeusFigendi** [devnope/ansible-workadventure#6]
Hi, thanks for your work here.

TL;DR: I don't know ansible well and I don't know how to fill/use the `vars.vault.yml` can you explain the variables?

The long story:

I am trying to get a work-adventure running for several months now (some hours per weekend you know) but I'm not very successful with this. Fortunately yesterday I found [your talk on the  R2R](https://media.ccc.de/v/divoc-reboot-to-respawn-2021-39-workadventure-selbst-betreiben-ein-serviervorschlag) and that sounded quite promising.
So I visited you repository here I tried to follow your instructions, I rented a small cloud server at Hetzner, booked a storage and a domain too. I installed `git ansible docker docker-compose` and before I configured the domain I tried to do this:

> Then you should create a Vault-File for Ansible and name it "vars.vault.yml". An example is provided in "vault-template.yml".

And here I stopped, I have a vague idea what Ansible does but I didn't used it 'til today (btw. I'm not a professional!) and when I open that file I see some variables to define but I am not sure at all what are their meanings.

In detail:

````
---
users:
  <your-user>:
    username: "<your-user"
    pass: "<passphrase-for-your-user-just-to-be-safe>"
    shell: "/usr/bin/bash"
    pubkey: "<your-ssh-pubkey>"
````

So is `<your-user>` the user **I** use to access the system or is it the user who will run the workadventure? And should it already exist or will ansible create it? I guess `<your-user>` and `<your-user` should contain the same value!? And finally: The `pubkey` is which? That public key the server I am using generated? My own from the machine I access the system? Or is it the users key who runs the workadventure and I have to generate it as that user?

````
local_become_pass: "set-your-password-here-so-you-do-not-have-to-interactively-enter-it"
````

Here again: Whos password? My? The new users? root?

````
letsencrypt:
  mail: "your-mail-address-for-letsencrypt-usage"
````

Okay this is clear to me :grin: 

````
#foradvancedjitsiiusage
docker:
  workadventure:
    secret:
      key: ""
    jitsi:
      key: ""
  coturn:
    username: "workadventure"
    pass: "passphrase-for-your-user-just-to-be-safe"
````

Okay since this is only "for advanced jitsi usage" I guess there is no need I understand this at the moment and I can leave this blank as long my jitsi servers are open.

Sorry bothering you in the issues but I didn't knew another way to contact you. Plus: Others might also have such questions, so maybe it's useful for those too :grin: 

Thank you.

Oh btw. in the end: Where should I drop the finished `vars.vault.yml`? Where I cloned the repository? (so in the example in `/opt/ansible-workadventure`) Or maybe in the home directory of the workadventure-user? Or is there a special place for ansible vaults? Sorry I really didn't ever use Ansible before :)

**Comment by devnope** [comment]
Hello,
`<your-user>` is you. I used this, because there are often cases, where there are more than one user or you have restrictions in your domain.
e.g. 
a user simon needs to have the username u1015 or something. you then can define the user and have a speaking name in the deployment.yml

The `pubkey` could be the one, you used at generating the vm or not... that's up to you. Imagine, you are creating the vm, but someone else will be managing it. or you have a ssh-key for root access and a ssh-key for user access.

`local_become_pass` is the root pass, since you don't know the one hetzner set. So you are doing it manually.

Where you drop the vault is completely up to you. Let it live, where it feels feel comfortable.

**Comment by devnope** [comment]
Does this answer your questions.

**Comment by DeusFigendi** [comment]
Maybe. So `<your-user>` is no user of the linux system but an independent user just in the softwarestack. Like the admin-account in a bullitin board or something like this?

Okay however I will try to do so and tell you if it worked or not. Thanks so far.

**Comment by DeusFigendi** [comment]
Well I guess that was fine, now I have trouble with the firewall settings but that's not about this issue.
